src/com/goide/highlighting/GoHighlightExitPointsHandlerFactory.java

```diff
@@ -85,12 +88,22 @@ public void visitCallExpr(@NotNull GoCallExpr o) {
     public static MyHandler createForElement(@NotNull Editor editor, PsiFile file, PsiElement element) {
       GoTypeOwner function = PsiTreeUtil.getParentOfType(element, GoFunctionLit.class, GoFunctionOrMethodDeclaration.class);
       if (function == null) return null;
-      if (element instanceof LeafPsiElement && ((LeafPsiElement)element).getElementType() == GoTypes.RETURN || isPanicCall(element)) {
+      if (shouldCreateMyHandler(element)) {
         return new MyHandler(editor, file, element, function);
       }
       return null;
     }
 
+    private static boolean shouldCreateMyHandler(PsiElement element) {
+      if (element instanceof LeafPsiElement) {
+        LeafPsiElement leaf = (LeafPsiElement)element;
+        return leaf.getElementType() == GoTypes.RETURN || leaf.getElementType() == GoTypes.FUNC || isPanicCall(leaf);
+      }
+      else {
```

Comment:
Do you mean that this method can be simplified to one big return statement?

I think it is more readable to have it as an if-else satatement